generator.rb

```diff
@@ -0,0 +1,111 @@
+def is_vowel(letter)
+  case letter
+  when "a", "e", "i", "o", "u"
+    return "true"
+  else
+    return "false"
+  end
+end
+
+def rev_str(strg)
+  aa = strg.split("")
+  rev = Array.new
+  aa.length.times do
+    rev << aa.pop
+  end
+  rev = rev.join
+  return rev
+end
+
+def check_length(word)
+  if word.length < 2
+    until word.length > 1
+      puts "Your word must be 2 or more characters."
+      word = gets.chomp
+    end
+    return word
+  else
+    return word
+  end
+end
+
+def vowel_index(word)
+  count = 0
+  word.each_char { |letter|
+    if is_vowel(letter) == "false"
+      count += 1
+    else
+      break
+    end
+  }
+  return count
+end
+
+def check_contain_vowel(word)
+  count = vowel_index(word)
+  if count == word.length
+    until count != word.length
+      puts "Word must contain a vowel."
+      word = gets.chomp
+      count = vowel_index(word)
+    end
+  end
+  return word
+end
+
+def run_generator
+  #----get first and second word from caller----
+  puts "What is the first word to combine?"
+  a = gets.chomp.downcase
+  a = check_length(a)
+  a = check_contain_vowel(a)
+  # get second word
+  puts "What is the second word to combine?"
+  b = gets.chomp.downcase
+  b = check_length(b)
+  b = check_contain_vowel(b)
```

Comment:
I love the idea of breaking these individual tasks out into separate methods - that makes the flow of this code very easy to understand. Good work!

If you wanted to go even further, you could combine these 4 repeated lines into another method, maybe something like `get_user_input`.